### PR TITLE
refactor: reduce excessive V3 API requests

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

No ticket, just a bandaid while trying to decrease our load on the V3 API.

- Hardly anything used the prediction's `shape_id`, so stop fetching the trip to populate this every time we parse a Vehicle.
- Stop fetching the parent stop ID every time we parse a Vehicle. Account for this in the various places this stop ID is retrieved, sometimes fetching the parent stop there if necessary.
- When fetching predictions, try to fetch by route ID and direction ID only. Then we can more effectively cache the result and use it in many situations, instead of fetching by many more unique combinations of stop/route pattern/route/directions.
- Stop caching the frequently-called `Dotcom.RealtimeSchedule.stop_data/3` function since it's mostly redundant, as it relies on many functions that are already configured with their own caching.

Running locally everything seems to be working OK. I found it hard to run the test suite locally, as I was sometimes getting rate limited before completion... so I'm curious to see how the tests work out here.

### CODE REVIEW

**Comment by kotva006** [comment]
> I was sometimes getting rate limited before completion... so I'm curious to see how the tests work out here.

Does this imply we are hitting the API more then after these changes?

File: lib/predictions/repo.ex [hunk 1]
```diff
@@ -43,11 +43,8 @@ defmodule Predictions.Repo do
   defp add_all_optional_params(opts) do
     @default_params
     |> add_optional_param(opts, :route)
-    |> add_optional_param(opts, :stop)
     |> add_optional_param(opts, :direction_id)
     |> add_optional_param(opts, :trip)
-    |> add_optional_param(opts, :route_pattern)
-    |> add_optional_param(opts, :"page[limit]")
```
[the remark this answers is not in the document]
Comment (thecristen) [reply lib/predictions/repo.ex:50]: The only place that was using it was removed in this PR.

image

**Comment by thecristen** [comment]
> Does this imply we are hitting the API more then after these changes?

@kotva006 Hard to tell since I hadn't run the tests locally from the master branch in a while. This PR's changes should result in us hitting the API less compared to current master branch (though it's hard to assess precisely since we don't have those metrics yet)

**Review by kotva006 — APPROVED** [review]
👍

**Pull request merged**